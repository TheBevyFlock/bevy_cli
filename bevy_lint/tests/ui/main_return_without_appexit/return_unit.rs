#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::main_return_without_appexit)]

use bevy::prelude::*;

fn main() -> () {
    //~^ HELP: try
    App::new().run();
    //~^ ERROR: an entrypoint that calls `App::run()` does not return `AppExit`
}
