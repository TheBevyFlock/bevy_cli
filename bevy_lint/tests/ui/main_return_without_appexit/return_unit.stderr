error: an entrypoint that calls `App::run()` does not return `AppExit`
  --> tests/ui/main_return_without_appexit/return_unit.rs:10:16
   |
8  | fn main() -> () {
   |              -- help: try: `AppExit`
9  |
10 |     App::new().run();
   |                ^^^^^
   |
   = note: `App::run()` returns `AppExit`, which can be used to determine whether the app exited successfully or not
note: the lint level is defined here
  --> tests/ui/main_return_without_appexit/return_unit.rs:4:9
   |
4  | #![deny(bevy::main_return_without_appexit)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

